### ISSUE TITLE
Unordered Funnels Quirks

### ISSUE THREAD

**Issue by neilkakkar** [PostHog/posthog#5694]
## Bug description

The step names don't make sense in unordered funnels:

![image](https://user-images.githubusercontent.com/7115141/130480589-058f3681-de59-4530-b949-21fcde8451a5.png)

If I switch ordering of steps around in an unordered funnel, it doesn't mean the step names change. Maybe we should have names like "Did 1 event", "Did 2 events", "Did 3 events". (ugh, desperately need better naming lol). cc: @paolodamico , @clarkus ?

Secondly, perhaps it doesn't make sense to allow choosing duplicate events in unordered funnels (i.e. 2 pageviews? What does this mean?) Do we want to disallow this?

**Comment by paolodamico** [comment]
Very related to [what a user reported today](https://posthog.slack.com/archives/C01G8S5T08Z/p1629769687117800). Agreed on disallowing repeated steps, and yeah I think changing the title to did 1 event, did 2 events, ... will add a ton of clarity around what this funnel actually means. We should also disallow reordering I guess?

**Comment by clarkus** [comment]
> Very related to [what a user reported today](https://posthog.slack.com/archives/C01G8S5T08Z/p1629769687117800). Agreed on disallowing repeated steps, and yeah I think changing the title to did 1 event, did 2 events, ... will add a ton of clarity around what this funnel actually means. We should also disallow reordering I guess?

Disallowing repeated steps is going to conflict with our default behavior when adding subsequent steps. We duplicate the previous entry, so we'll be defaulting to an invalid state when using "any" ordering. 

I think the place we need to clarify is in the funnel itself, not so much in the query building component. Changing the labels makes sense, but we might also gain some value from restating the query parameters in a summary above the funnel. I'll spend some time on this. Short term, a quick update to @neilkakkar's suggestion sounds good to me.

**Comment by paolodamico** [comment]
Well IIRC your newer designs already include a solution for this with the duplicate button right? Perhaps we could start with this new behavior in funnels. Sounds good on everything else @clarkus !

**Comment by akbansa** [comment]
The funnels are basically a way to see how many people are moving from A to B via C, D...etc.

Now any order for any business would not make sense because marketers, PMs are sure about where a person is starting and where we want them to land, i.e. a conversion.
Note: I personally do not use something like any order.


**One scenario**: where the order of intermediate steps may not matter in different user journeys, so the order of intermediate steps may not matter for me in many cases. Since the order of the intermediate steps need not be in the same order, so with posthog we can create a custom event (composite action) either (C) or (D) happened, the funnel A => C|D => B can be created.

**Comment by neilkakkar** [comment]
@akbansa : To answer your question about the conversion rate decreasing. We can't actually know beforehand if the rate would increase or decrease with "any order". 

This is because the absolute values of both step 1 and step 3 will be greater than, or equal to the old values (sequential order). In this case, since people who did 1 event only increases, the conversion % goes down.

**Comment by clarkus** [comment]
> Well IIRC your newer designs already include a solution for this with the duplicate button right? Perhaps we could start with this new behavior in funnels. Sounds good on everything else @clarkus !

@paolodamico, the newer designs do provide distinct actions for adding and duplicating, but they also expect a bit more available width for the query builder. I think it'll work but we might have to adjust some container proportions in order to make things align.

**Comment by clarkus** [comment]
Here's a quick update to make the "any" evaluation mode a bit more obvious in the visualization. I've dropped numbers here and instead I am showing a placeholder icon. Step names have been changed to `Completed N step(s)`. 

![C stacked-1](https://user-images.githubusercontent.com/254612/130688491-32d200a6-8fcf-4cc5-9897-3b7df485e91c.png)
![C stacked](https://user-images.githubusercontent.com/254612/130688494-64304631-0ae6-42a4-a12a-89f2d840cddb.png)
https://www.figma.com/file/9yWtngNb1AIuf6KmXaEPJA/App-doodles?node-id=1123%3A145127
